src/build-articles.js

```diff
@@ -0,0 +1,255 @@
+// Import Node.js Dependencies
+import fs from "node:fs";
+import path from "node:path";
+
+// Import Third-party Dependencies
+import { marked } from "marked";
+import createDOMPurify from "dompurify";
+import { JSDOM } from "jsdom";
+
+const inputDir = "./articles";
+const outputDir = "./blog";
+
+// processes all markdown files in articles directory
+export function convertAllMarkdownArticles(authors) {
+  if (!fs.existsSync(inputDir)) {
+    throw new Error(`${inputDir} directory does not exist`);
+  }
+
+  const files = fs.readdirSync(inputDir).filter((file) => file.endsWith(".md"));
+
+  const articles = files.map((file) => {
+    const filePath = path.join(inputDir, file);
+
+    return generateBlogPost(filePath, outputDir);
```

Comment:
```suggestion
    return generateBlogPost(filePath);
```
since outputDir is a global variable no need to pass it to generateBlogPost who doesn't expect this param